flink-table-store-flink/flink-table-store-flink-common/src/test/java/org/apache/flink/table/store/connector/action/MergeIntoActionITCase.java

```diff
@@ -130,57 +131,78 @@ public void testVariousChangelogProducer(
-                        changelogRow("+U", 3, "v_3_nmu", "not_matched_upsert", "02-27"),
-                        changelogRow("-D", 5, "v_5", "creation", "02-28"),
-                        changelogRow("-D", 6, "v_6", "creation", "02-28"),
-                        changelogRow("-D", 9, "v_9", "creation", "02-28"),
-                        changelogRow("-D", 10, "v_10", "creation", "02-28")),
+                        changelogRow("-D", 8, "v_8", "creation", "02-28")),
                 Arrays.asList(
                         changelogRow("+I", 1, "v_1", "creation", "02-27"),
-                        changelogRow("+U", 2, "v_2_nmu", "not_matched_upsert", "02-27"),
-                        changelogRow("+U", 3, "v_3_nmu", "not_matched_upsert", "02-27"),
-                        changelogRow("+U", 7, "Seven", "matched_upsert", "02-28"),
-                        changelogRow("+I", 8, "v_8", "insert", "02-29"),
-                        changelogRow("+I", 11, "v_11", "insert", "02-29")));
+                        changelogRow("+I", 2, "v_2", "creation", "02-27"),
+                        changelogRow("+I", 3, "v_3", "creation", "02-27"),
+                        changelogRow("+I", 5, "v_5", "creation", "02-28"),
+                        changelogRow("+I", 6, "v_6", "creation", "02-28"),
+                        changelogRow("+I", 7, "v_7", "creation", "02-28"),
+                        changelogRow("+I", 9, "v_9", "creation", "02-28"),
+                        changelogRow("+I", 10, "v_10", "creation", "02-28")));
+    }
+
+    @Test
+    public void testUsingDdlSource() throws Exception {
+        // prepare table T
+        prepareTable(CoreOptions.ChangelogProducer.NONE);
+
+        TestValuesTableFactory.registerData(
+                Arrays.asList(
+                        changelogRow("+I", 1, "v_1", "02-27"),
+                        changelogRow("+I", 4, null, "02-27"),
+                        changelogRow("+I", 8, null, "02-28")));
+
+        String catalog =
+                String.format(
+                        "CREATE CATALOG test_cat WITH ('type' = 'table-store', 'warehouse' = '%s')",
+                        getTempDirPath());
+        String useCatalog = "USE CATALOG test_cat";
+        String id =
+                TestValuesTableFactory.registerData(
+                        Arrays.asList(
+                                changelogRow("+I", 1, "v_1", "02-27"),
+                                changelogRow("+I", 4, null, "02-27"),
+                                changelogRow("+I", 8, null, "02-28")));
+        String ddl =
+                String.format(
+                        "CREATE TEMPORARY TABLE S (k INT, v STRING, dt STRING)\n"
+                                + "WITH ('connector' = 'values', 'bounded' = 'true', 'data-id' = '%s');",
+                        id);
+
+        MergeIntoAction action = new MergeIntoAction(warehouse, database, "T");
+        action.withSqlSource("test_cat.default.S", catalog, useCatalog, ddl)
```

Comment:
`test_cat.default.S` -> `S`.
Add comment: test current catalog and current database.